### PR TITLE
Removed most of the channels related code

### PR DESCRIPTION
**Pull request by k-antipochkin**
### Description of Changes

Removed the dead code related to channels which are no longer in use

### Breaking Changes

-   None

### Release Checklist

Prepare:

-   ~~[ ] Detail any breaking changes. Breaking changes require a new major version number~~

Bump versions in:

-   ~~[ ] package.json~~
-   ~~[ ] package-lock.json (run `npm install` after editing package.json)~~
-   ~~[ ] src/core/index.ts~~

Build:

-   [X] `npm run dist`
-   [X] Commit resulting `dist/` changes

Release:

-   [ ] Squash and merge to master
-   [ ] Delete branch once merged
-   [ ] Create tag from master matching chosen version
-   [ ] Fill out release notes
-   [ ] Run `npm publish --access public`
-   [ ] Publish `dist/worker.js` and `dist/web-bundle.js` to CDN (re-run `npm run dist` locally to update `web-bundle.js`)

### CODE REVIEW

File: src/prompts/index.tsx [hunk 1]
```diff
@@ -126,22 +110,10 @@ export class PromptManager {
...
         }
     };
 
-    private onPostActionConfirm = async (
-        prompt: PushPromptConfig,
-        channelSelections?: ChannelListItem[]
-    ) => {
-        await this.handleUserChannelSelection(channelSelections);
-
-        this.setState('ready');
```
Comment (cgwyllie) [inline src/prompts/index.tsx:141]: Is this state transition back to ready now missing?

Do we still need to keep the 'hideAndSuppressPrompts' below as well?

Comment (cgwyllie) [reply src/prompts/index.tsx:141]: (perhaps not, this is about the 'post action', which if I read correctly, we have removed the concept/handling of 'post actions'?)

Comment (k-antipochkin) [reply src/prompts/index.tsx:141]: Yes, following the code the currentPostAction is always null so it never renders the ChannelsDialog which called the onPostActionConfirm. 

Perhaps it was half deleted in [this PR](https://github.com/Kumulos/KumulosSdkWeb/pull/57/files)

**Review by cgwyllie — APPROVED** [review]

File: types/src/core/channels.d.ts [hunk 2]
```diff
@@ -1,47 +1 @@
...
-     */
-    listChannels(): Promise<Channel[]>;
-    /**
-     * Creates a push channel and optionally subscribes the current installation.
-     *
-     * Name is optional, but required if showInPortal is true.
-     */
-    createChannel(channelSpec: ChannelSpec): Promise<Channel>;
-}
+export {};
```
Comment (vvoicehovics) [inline types/src/core/channels.d.ts]: can delete the file?

File: src/prompts/index.tsx [hunk 3]
```diff
@@ -46,8 +43,6 @@ export class PromptManager {
     private pushOpsManager?: PushOpsManager;
     private ui?: Ui;
     private prompts: PromptConfigs<PushPromptConfig>;
-    private currentPostAction?: PromptAction;
```
Comment (vvoicehovics) [inline src/prompts/index.tsx:49]: can remove remaining references to 'postaction' ?

File: src/core/index.ts [hunk 4]
```diff
@@ -1,8 +1,6 @@
 import { performFetch, cyrb53, uuidv4 } from './utils';
 import { del, get, set } from './storage';
 
-import { Channel } from './channels';
```
Comment (vvoicehovics) [inline src/core/index.ts:4]: in this file and few others few references to 'channel', e.g. ChannelSubAction. not used anymore?

Comment (k-antipochkin) [reply src/core/index.ts:4]: Probably not used but their usage is not trivial to track to safely remove. Will require some more effort, will do in another PR

File: types/src/core/channels.d.ts (hunk 2 above)
Comment (k-antipochkin) [reply types/src/core/channels.d.ts]: Done

File: src/prompts/index.tsx (hunk 3 above)
Comment (k-antipochkin) [reply src/prompts/index.tsx:49]: Removed

**Pull request merged**